docs/book/ceph-authorization.md

```diff
@@ -0,0 +1,1197 @@
+kubectl create -f rgw-external.yaml
+```
+
+Check that both the `internal` and `external` RGW services are `Running`.
+
+```bash
+kubectl -n rook-ceph get service rook-ceph-rgw-my-store rook-ceph-rgw-my-store-external
+```
+
+### 7. Create Object Store Users
+
+Create two object store users `Alice` and `Bob`.
+
+**object-user-alice.yaml**
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephObjectStoreUser
+metadata:
+  name: alice
+  namespace: rook-ceph
+spec:
+  store: my-store
+  displayName: "Alice"
+```
+
+**object-user-bob.yaml**
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephObjectStoreUser
+metadata:
+  name: bob
+  namespace: rook-ceph
+spec:
+  store: my-store
+  displayName: "Bob"
+```
+
+Now create the users.
+
+```bash
+kubectl create -f object-user-alice.yaml
+kubectl create -f object-user-bob.yaml
+```
+
+When the object store user is created the Rook operator will create the RGW user on the object store `my-store`, and store the user's Access Key and Secret Key in a Kubernetes secret in the namespace `rook-ceph`.
+
+### 8. Understanding the OPA policy
+
+As we saw earlier, the OPA spec contained a ConfigMap that defined the policy to be used to authorize requests received by the `Ceph Object Gateway`. Below is the policy:
+
+**authz.rego**
+
+{%ace lang='python'%}
+package ceph.authz
+
+default allow = false
+
+#-----------------------------------------------------------------------------
+# Data structures containing location info about users and buckets.
+# In real-world deployments, these data structures could be loaded into
+# OPA as raw JSON data. The JSON data could also be pulled from external
+# sources like AD, Git, etc.
+#-----------------------------------------------------------------------------
+
+# user-location information
+user_location = {
+    "alice": "UK",
+    "bob":   "USA"
+}
+
+# bucket-location information
+bucket_location = {
+    "supersecretbucket": "USA"
+}
+
+# Allow access to bucket in same location as user.
+allow {
+    input.method = "HEAD"
+    is_user_in_bucket_location(input.user_info.user_id, input.bucket_info.bucket.name)
+}
+
+allow {
+    input.method = "GET"
+}
+
+allow {
+    input.method = "PUT"
+    input.user_info.display_name = "Bob"
+}
+
+allow {
+    input.method = "DELETE"
+    input.user_info.display_name = "Bob"
+}
+
+# Check if the user and the bucket being accessed belong to the same
+# location.
+is_user_in_bucket_location(user, bucket) {
+    user_location[user] == bucket_location[bucket]
+}
+{%endace%}
+
+**The above policy will restrict a user from accessing a bucket whose location does not match the user's location.**. The user's and bucket's location is hardcoded in the policy for simplicity and in the real-world can be fetched from external sources or pushed into OPA using it's REST API.
+
+In tha above policy, `Bob's` location is `USA` while `Alice's` is `UK`. Since the bucket `supersecretbucket` is located in the `USA`, `Alice` should not be able to access it.
```

Comment:
typo
```suggestion
In the above policy, `Bob's` location is `USA` while `Alice's` is `UK`. Since the bucket `supersecretbucket` is located in the `USA`, `Alice` should not be able to access it.
```